trojsten/special/plugin_prask_5_1_1/core.py

```diff
@@ -21,42 +21,164 @@ def run(cls, x, try_count):
+    TABLE = [
+        '-',
+        'H',
+        'He',
+        'Li',
+        'Be',
+        'B',
+        'C',
+        'N',
+        'O',
+        'F',
+        'Ne',
+        'Na',
+        'Mg',
+        'Al',
+        'Si',
+        'P',
+        'S',
+        'Cl',
+        'Ar',
+        'K',
+        'Ca',
+        'Sc',
+        'Ti',
+        'V',
+        'Cr',
+        'Mn',
+        'Fe',
+        'Co',
+        'Ni',
+        'Cu',
+        'Zn',
+        'Ga',
+        'Ge',
+        'As',
+        'Se',
+        'Br',
+        'Kr',
+        'Rb',
+        'Sr',
+        'Y',
+        'Zr',
+        'Nb',
+        'Mo',
+        'Tc',
+        'Ru',
+        'Rh',
+        'Pd',
+        'Ag',
+        'Cd',
+        'In',
+        'Sn',
+        'Sb',
+        'Te',
+        'I',
+        'Xe',
+        'Cs',
+        'Ba',
+        'La',
+        'Ce',
+        'Pr',
+        'Nd',
+        'Pm',
+        'Sm',
+        'Eu',
+        'Gd',
+        'Tb',
+        'Dy',
+        'Ho',
+        'Er',
+        'Tm',
+        'Yb',
+        'Lu',
+        'Hf',
+        'Ta',
+        'W',
+        'Re',
+        'Os',
+        'Ir',
+        'Pt',
+        'Au',
+        'Hg',
+        'Tl',
+        'Pb',
+        'Bi',
+        'Po',
+        'At',
+        'Rn',
+        'Fr',
+        'Ra',
+        'Ac',
+        'Th',
+        'Pa',
+        'U',
+        'Np',
+        'Pu',
+        'Am',
+        'Cm',
+        'Bk',
+        'Cf',
+        'Es',
+    ]
+    TARGET = "Al,I,Ca"
 
     @classmethod
     def run(cls, x, try_count):
-        a, b = 0, 1
-        for i in range(x):
-            a, b = b, a + b
-            if len(str(a)) > 30:
-                return 'PRILIS VELKE'
-        return str(a)
+        str_x = str(x)
+        elements = []
+        start = 0
+        if len(str_x) % 2 == 1:
+            elements.append(cls.TABLE[int(str_x[0])])
+            start = 1
+        for i in range(start, len(str_x), 2):
+            elements.append(cls.TABLE[int(str_x[i:i + 2])])
+        return ','.join(elements)
 
 
 class Level3(object):
 
-    TARGET = "3528"
+    TARGET = "7907"
+    PRIMES = []
+
+    @classmethod
+    def generate_primes(cls):
+        limit = 10**6
+        erat = [True] * limit
+        for i in range(2, limit):
+            if not erat[i]:
+                continue
+            cls.PRIMES.append(i)
+            for j in range(i, limit, i):
+                erat[j] = False
 
     @classmethod
     def run(cls, x, try_count):
-        new_x = x
-        if x % 2 == 0:
-            new_x *= 3
-        if x % 3 == 0:
-            new_x *= 2
-        return str(new_x)
+        if not cls.PRIMES:
+            cls.generate_primes()
+        if x >= len(cls.PRIMES):
+            return 'PRILIS VELKE'
+        return str(cls.PRIMES[x])
 
 
 class Level4(object):
 
-    TARGET = "3684591"
+    TARGET = "Oto"
+    DATA = []
+    FILE = os.path.join(os.path.dirname(__file__), "mena.txt")
```

Comment:
Extract-núť `mena.txt`. Taktiež, je to dobrý nápad takto brať súbory?

Reply:
Hej, je lebo to neni zbytocne v kode, robi to tak aj sysel minule roky.

Reply:
To že to Syseľ tak robil minulé roky neznamená, že to je v pohode.